nifi-toolkit/nifi-toolkit-admin/src/main/groovy/org/apache/nifi/toolkit/admin/client/NiFiClientFactory.groovy

```diff
@@ -116,22 +121,27 @@ class NiFiClientFactory implements ClientFactory{
         return sslContext;
     }
 
-    private static class NiFiHostnameVerifier implements HostnameVerifier {
+    static class NiFiHostnameVerifier implements HostnameVerifier {
 
         @Override
         public boolean verify(final String hostname, final SSLSession ssls) {
             try {
-                for (final Certificate peerCertificate : ssls.getPeerCertificates()) {
-                    if (peerCertificate instanceof X509Certificate) {
-                        final X509Certificate x509Cert = (X509Certificate) peerCertificate;
-                        final List<String> subjectAltNames = getSubjectAlternativeNames(x509Cert);
-                        if (subjectAltNames.contains(hostname.toLowerCase())) {
-                            return true;
-                        }
+
+                final Certificate peerCertificate = ssls.getPeerCertificates()[0]
+                final X509Certificate x509Cert = CertificateUtils.convertAbstractX509Certificate(peerCertificate)
+                final String dn = x509Cert.getSubjectDN().getName().trim()
+                final String cn = dn.tokenize(",").find { cn -> cn.startsWith("CN=")}
```

Comment:
Unfortunately I think this could be susceptible to situations where the CN contains a `,` -- `CN=notmysite.com,mysite.com, OU=MySite, C=US` would tokenize into `CN=notmysite.com`, `mysite.com`, ` OU=MySite`, and ` C=US`. 

See this [StackOverflow answer](http://stackoverflow.com/a/7933650/70465) for a code sample using `javax.naming.ldap.LdapName` to parse the DN into "RDN"s (*Relative Distinguished Name*, or the component terms of a DN).